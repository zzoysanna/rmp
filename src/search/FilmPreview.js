import React, { Component } from 'react';
import './styles/film-preview.less';
import { Link } from "react-router-dom";
import * as actions from "../store/actions";
import { connect } from "react-redux";

class FilmPreview extends Component {
	prepareDate(date) {
		const releaseDate = new Date(date);
		return releaseDate.getFullYear();
	}

	render() {
		const { title, release_date, genres, poster_path, id } = this.props.film;
		const {onSetCurrentFilm} = this.props;
		return (
			<Link className="film" to={`/film/${id}`} onClick={() => onSetCurrentFilm(id)}>
				<div style={{ backgroundImage: `url(${poster_path})` }} className="poster">&nbsp;</div>
				<div className="info">
					<p>
						<span className="title">{title}</span>
						<span className="genre">{genres.join('&')}</span>
					</p>
					<span className="year">{this.prepareDate(release_date)}</span>
				</div>
			</Link>
		);
	}
}

const mapStateToProps = (state) => {
	return {}
};

const mapDispatchToProps = (dispatch) => {
	return {
		onSetCurrentFilm: (id) => dispatch(actions.setCurrentFilm(id)),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(FilmPreview);

