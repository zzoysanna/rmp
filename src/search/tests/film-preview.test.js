import React from 'react';
import { shallow } from 'enzyme';
import FilmPreview from '../FilmPreview';

const film1 = {
	id: 1,
	title: 'test-title-1',
	poster_path: 'test-path-1',
	genres: ['a', 'b'],
	release_date: '1234',
};

describe('FilmPreview', () => {
	let film;

	beforeAll(() => {
		film = film1;
	});

	it('should be render correctly', () => {
		const component = shallow(<FilmPreview film={film}/>);
		expect(component).toMatchSnapshot();
	});
});
