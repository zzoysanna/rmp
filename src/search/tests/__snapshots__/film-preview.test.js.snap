// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPreview should be render correctly 1`] = `ShallowWrapper {}`;
